packages/runtime-core/src/main/kotlin/elide/runtime/core/PolyglotEngineConfiguration.kt

```diff
@@ -31,6 +33,77 @@ import elide.runtime.core.PolyglotEngineConfiguration.HostAccess.ALLOW_NONE
     ALLOW_NONE,
   }
 
+  /**
+   * Provides read-only information about the Host platform, which can be used by plugins to run platform-specific
+   * code.
+   */
+  public data class HostPlatform(
+    public val os: OperatingSystem,
+    public val arch: Architecture,
+  ) {
+    /** Enumerates the Operating System families supported by the runtime */
+    public enum class OperatingSystem {
+      LINUX,
+      DARWIN,
+      WINDOWS,
+    }
+
+    /** Enumerates the architectures supported by the runtime. */
+    public enum class Architecture {
+      AMD64,
+      ARM64,
+    }
+
+    override fun toString(): String {
+      // provide a more compact representation, such as 'linux_amd64'
+      return "${os.name.lowercase()}$STANDARD_SEPARATOR${arch.name.lowercase()}"
+    }
+
+    public companion object {
+      private const val STANDARD_SEPARATOR = '_'
+
+      /** Parse a generic OS name, such as the value obtained from the 'os.name' system property. */
+      public fun parseOperatingSystem(name: String): OperatingSystem = when {
+        name.contains("linux") -> OperatingSystem.LINUX
+        name.contains("windows") -> OperatingSystem.WINDOWS
+        name.contains("mac") || name.contains("darwin") -> OperatingSystem.DARWIN
+        else -> error("Unrecognized Operating System family: $name")
+      }
+
+      /** Parse a generic architecture name, such as the value obtained from the 'os.arch' system property. */
+      public fun parseArchitecture(name: String): Architecture = when {
+        name.contains("x86_64") || name.contains("amd64") -> Architecture.AMD64
+        name.contains("arm64") || name.contains("aarch64") -> Architecture.ARM64
+        else -> error("Unrecognized architecture: $name")
+      }
+
+      /** Parse a well-formed [HostPlatform] value, as obtained from [HostPlatform.toString]. */
+      public fun parsePlatform(name: String): HostPlatform {
+        val separatorIndex = name.indexOf(STANDARD_SEPARATOR)
+        require(separatorIndex != -1) { "Invalid platform name: $name" }
+
+        return HostPlatform(
+          os = parseOperatingSystem(name.substring(startIndex = 0, endIndex = separatorIndex)),
+          arch = parseArchitecture(name.substring(startIndex = separatorIndex + 1)),
+        )
+      }
+
+      /**
+       * Resolve information about the Host, such as OS and arch. This info is used by language plugins to load
+       * platform-specific resources.
+       */
+      public fun resolve(): HostPlatform {
+        return HostPlatform(
+          os = HostPlatform.parseOperatingSystem(System.getProperty("os.name", "unknown").lowercase()),
+          arch = HostPlatform.parseArchitecture(System.getProperty("os.arch", "unknown").lowercase()),
+        )
+      }
+    }
+  }
+
+  /** Information about the platform hosting the runtime. */
+  public val hostPlatform: HostPlatform = HostPlatform.resolve()
```

Comment:
### Host Platform info

The `PolyglotEngineConfiguration` now exposes cached information about the platform hosting the runtime. This allows plugins to select embedded bundles that match the current platform without handling OS or arch resolution manually.

The `HostPlatform` class and its companion provide functions to resolve host information, and to parse individual components from generic sources (mainly `os.name` and `os.arch` JVM System Properties).